internal/daemon/daemon.go

```diff
@@ -492,6 +495,9 @@ func (d *Daemon) StartAPI(bootstrap bool, initConfig map[string]string, newConfi
 
 // ClusterCert ensures both the daemon and state have the same cluster cert.
 func (d *Daemon) ClusterCert() *shared.CertInfo {
+	d.clusterMu.RLock()
```

Comment:
Since the pointer to `d.clusterCert` is returned, wouldn't this break the logic of the read lock if the caller reads from the pointers address afterwards?

Reply:
Good catch, thanks